src/main/java/org/folio/inventory/validation/ItemUpdateValidator.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.inventory.validation;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import io.vertx.ext.web.RoutingContext;
+
+public final class ItemUpdateValidator {
+
+  private ItemUpdateValidator() {}
+
+  /**
+   * Validation logic for {@link org.folio.inventory.resources.Items#update(RoutingContext)}
+   * method.
+   *
+   * @param oldItem - The item from DB.
+   * @param newItem - The item from request.
+   * @return An optional with validation error, if any.
+   */
+  public static Optional<ValidationError> validateItemUpdate(Item oldItem, Item newItem) {
+    ValidationError validationError = null;
+
+    if (!Objects.equals(newItem.getHrid(), oldItem.getHrid())) {
+      validationError = new ValidationError(
+        "HRID can not be updated", "hrid", newItem.getHrid());
+    }
+
+    if (claimedReturnedItemMarkedMissing(oldItem, newItem)) {
+      validationError = new ValidationError(
+        "Claimed returned item can not be marked as missing",
+        "status.name", ItemStatusName.MISSING.value());
+    }
```

Comment:
I think it makes sense to try to separate these two cases into separate methods

Reply:
Do you mean to separate and execute them in this method, or in the resource's one?

My intent was to move the logic to separate class, because it grows and the `Items` class has already had ~1k lines.

Reply:
I was thinking separate methods within this class, to make that method read well at a higher level of abstraction.

I'm happy for it to be in this class (I think it needs a better name). Thank you for extracting this from the `Items` class.